tools/mksurfdata_map/src/mksurfdat.F90

```diff
@@ -1188,6 +1190,27 @@ program mksurfdat
 
           call change_landuse(ldomain, dynpft=.true.)
 
+          ! Make sure sum of all land cover types except natural vegetation 
+          ! does not exceed 100. If it does, subtract excess from these land cover
+          ! types proportionally.
+
+          do n = 1,ns_o
+             suma = pctlak(n) + pctwet(n) + pcturb(n) + pctgla(n) + pctcft(n)%get_pct_l2g()
+             if (suma > 250._r4) then
+                write (6,*) subname, ' error: sum of pctlak, pctwet,', &
+                     'pcturb, pctgla, and pctcrop is greater than 250%'
+                write (6,*)'n,pctlak,pctwet,pcturb,pctgla,pctcrop= ', &
+                     n,pctlak(n),pctwet(n),pcturb(n),pctgla(n),pctcft(n)%get_pct_l2g()
+                call abort()
+             else if (suma > 100._r4) then
+                pctlak(n) = pctlak(n) * 100._r8/suma
+                pctwet(n) = pctwet(n) * 100._r8/suma
+                pcturb(n) = pcturb(n) * 100._r8/suma
+                pctgla(n) = pctgla(n) * 100._r8/suma
+                call pctcft(n)%set_pct_l2g(pctcft(n)%get_pct_l2g() * 100._r8/suma)
+             end if
+          end do
```

Comment:
After reviewing the changes made by @keerzhang1 last week, I was beginning to come to the same conclusion.  My preferred approach would be to combine these three PRs into one PR (#1579 ), rather than in two steps. But will leave it up to you and Erik to decide.  As you've noted, there is also a fourth PR forthcoming from issue #1572 , code that reads in HASURBAN and initializes landunits in memory (which will now need to be changed to PCT_URBAN_MAX I guess).  I think that would come in as part of the feature additions if we separated out feature from behavior.

Reply:
I also like to idea to combine these three PRs into one PR (#1579) by the two steps proposed by Bill.  I think perhaps I can combine #1073 with #1579 first. Then after we finalize #1564, we can apply the changes in #1564 to #1579 (or a new PR) or keep them separate.

Reply:
Thanks @keerzhang1 for offering to combine #1073  with #1579 , that sounds good to me.  Let us know if you have any questions.  I'll work on responding to the review of #1564

Reply:
Thanks a lot @keerzhang1 ! I'm thinking it will be easiest if you use #1579 as a starting point, since that's already in good shape, and then bring in the relevant changes from #1073 into that. Once we look that over and are happy with that merge, then @olyson can bring his changes from this PR (#1564 ) on top of that.

A few notes when integrating the changes from #1073 
- There's no need to bring in the namelist changes from that PR
- For the text file that connects the perl script to the mksurfdata_map fortran code, you can use the approach you implemented for urban: using the same text file as for transient PFTs but adding an extra line for each year.
- You don't need to update the perl code to add these lake files for now, unless that will help you with your testing.

Please let us know if you have questions or run into any problems.